source/extensions/common/aws/signer_base_impl.h

```diff
@@ -24,6 +24,23 @@ class SignatureHeaderValues {
 
 using SignatureHeaders = ConstSingleton<SignatureHeaderValues>;
 
+class SignatureQueryParameterValues {
+public:
+  // Query string parameters require camel case
+  static constexpr char AmzAlgorithm[] = "X-Amz-Algorithm";
+  static constexpr char AmzCredential[] = "X-Amz-Credential";
+  static constexpr char AmzDate[] = "X-Amz-Date";
+  static constexpr char AmzRegionSet[] = "X-Amz-Region-Set";
+  static constexpr char AmzSecurityToken[] = "X-Amz-Security-Token";
+  static constexpr char AmzSignature[] = "X-Amz-Signature";
+  static constexpr char AmzSignedHeaders[] = "X-Amz-SignedHeaders";
+  static constexpr char AmzExpires[] = "X-Amz-Expires";
+  // Expiration time of query parameter request, in seconds
+  const uint16_t DefaultExpiration = 5;
+};
+
+using SignatureQueryParameters = ConstSingleton<SignatureQueryParameterValues>;
```

Comment:
If `static constexpr` is used, then I think you needn't the `ConstSingleton`.

Reply:
@wbpcode do you mean take these static constexpr out of the SignatureQueryParameterValues class entirely? ConstSingleton is how this class is instantiated.

Reply:
I mean if static `constexpr` is used, you may could access the value by `SignatureQueryParameterValues::xxxx` directly and needn't instance this class because all member are `static`?

Reply:
ty - i did not know this. will clean these up!